plugins/hls-ormolu-plugin/src/Ide/Plugin/Ormolu.hs

```diff
@@ -59,7 +75,7 @@ provider ideState typ contents fp _ = withIndefiniteProgress title Cancellable $
  where
    title = T.pack $ "Formatting " <> takeFileName (fromNormalizedFilePath fp)
 
-   ret :: Either OrmoluException T.Text -> Either ResponseError (List TextEdit)
+   ret :: Either SomeException T.Text -> Either ResponseError (List TextEdit)
```

Comment:
Why `SomeException` now? Doesn't this risk of swallowing async exceptions? IIRC, ghcide relies heavily on being able to interrupt actions, unless there is a good reason, we shouldn't catch `SomeException` since code will change and then eventually not do what we expect.

Reply:
> Why `SomeException` now? Doesn't this risk of swallowing async exceptions?

It does not, as we are only catching `OrmoluException` and `IOException`:
https://github.com/haskell/haskell-language-server/blob/e9f9a49014276eb2dceab0cb91faccf4d717f5e8/plugins/hls-ormolu-plugin/src/Ide/Plugin/Ormolu.hs#L66-L69

Previously, only `OrmoluException` was caught, but `getCabalInfoForSourceFile`/`getCabalInfoForSourceFile`, which are added in this PR, can alos throw `IOException`.

Reply:
Ah, my mistake. This is unfortunate, wish we had some union type here :(

Reply:
Yeah, I can change it to `Either (Either OrmoluException IOException) T.Text` if you prefer, but it indeed seems more clumsy :crying_cat_face:

Reply:
I don't think I do :D